### PR TITLE
feat!: add comprehensive reserved keywords for presto and redshift

### PR DESCRIPTION
**Pull request by tobymao**

### CODE REVIEW

File: sqlglot/parser.py [hunk 1]
```diff
@@ -1764,7 +1764,11 @@ def _parse_stored(self) -> exp.FileFormatProperty:
     def _parse_property_assignment(self, exp_class: t.Type[E], **kwargs: t.Any) -> E:
         self._match(TokenType.EQ)
         self._match(TokenType.ALIAS)
-        return self.expression(exp_class, this=self._parse_field(), **kwargs)
+        field = self._parse_field()
+        if isinstance(field, exp.Identifier) and not field.quoted:
+            field = exp.var(field)
```
Comment (georgesittas) [inline sqlglot/parser.py:1769]: what triggered this?

File: tests/dialects/test_spark.py [hunk 2]
```diff
@@ -234,7 +234,7 @@ def test_spark(self):
         self.validate_identity("first_value(col, true)", "FIRST_VALUE(col) IGNORE NULLS")
         self.validate_identity("last(col, true)", "LAST(col) IGNORE NULLS")
         self.validate_identity("last_value(col, true)", "LAST_VALUE(col) IGNORE NULLS")
-        self.validate_identity("DESCRIBE EXTENDED db.table")
+        self.validate_identity("DESCRIBE EXTENDED db.tbl")
```
Comment (georgesittas) [inline tests/dialects/test_spark.py:237]: how come these tests changed?

Comment (tobymao) [reply tests/dialects/test_spark.py:237]: tried adding them to spark but then a bunch of this stuff changed

File: sqlglot/parser.py (hunk 1 above)
Comment (tobymao) [reply sqlglot/parser.py:1769]: because we have some ddl that is like diststyle all, and all got quoted

**Review by georgesittas — APPROVED** [review]

**Pull request merged**